src/msha/handlers/dab.handler.spec.ts

```diff
@@ -0,0 +1,63 @@
+import http from "http";
+import { URL } from "url";
+import { injectHeaders, isDataApiRequest } from "./dab.handler";
+
+describe("isDataApiRequest", () => {
+  const req = { url: "/data-api/users" } as http.IncomingMessage;
+
+  it('returns true when the path starts with "/data-api/"', () => {
+    const result = isDataApiRequest(req);
+    expect(result).toEqual(true);
+  });
+
+  it('returns false when the path does not start with "/data-api/"', () => {
+    const req2 = { url: "/api/users" } as http.IncomingMessage;
+    const result = isDataApiRequest(req2);
+    expect(result).toEqual(false);
+  });
+
+  it("returns true when the path starts with a custom rewrite path", () => {
+    const req2 = { url: "/api/users" } as http.IncomingMessage;
+    const result = isDataApiRequest(req2, "/data-api/users");
+    expect(result).toEqual(true);
+  });
+});
+
+describe("injectHeaders", () => {
+  let req: http.ClientRequest;
+
+  beforeEach(() => {
+    req = {
+      getHeader: jest.fn(),
+      setHeader: jest.fn(),
+      path: "/example/path",
+    } as unknown as http.ClientRequest;
+  });
+
+  it("sets x-ms-original-url header if not already set", () => {
+    const host = "http://example.com";
+    injectHeaders(req, host);
+    const expectedUrl = encodeURI(new URL(req.path!, host).toString());
```

Comment:
shall we avoid using the same code in the test as well .. can we keep the encoded string here.. 

Also shall we see if we can have space or special character in the path.

Reply:
Yup